engine/verification/match/engine.go

```diff
@@ -448,9 +450,24 @@ func (e *Engine) handleChunkDataPack(originID flow.Identifier,
 	}
 
 	if sender.Role != flow.RoleExecution {
-		return engine.NewInvalidInputError("receives chunk data pack from a non-execution node")
+		return engine.NewInvalidInputError("received chunk data pack from a non-execution node")
+	}
+
+	// Check that the collection corresponds to the DataPack's CollectionID
+	if collection.ID() != chunkDataPack.CollectionID {
```

Comment:
I am generally wondering about the placement of this check. To me, it seems that it contains a special-case for the system chunk. However, I am generally concerned about simply basing this check on data the EN provides and could lie about: https://github.com/onflow/flow-go/blob/42c5833b218755cbe2b10300ca177445fbfe2f8e/engine/verification/match/engine.go#L463


Don't we have a check further down, where we determine whether we expect the chunk to be the system chunk or not (?): https://github.com/onflow/flow-go/blob/42c5833b218755cbe2b10300ca177445fbfe2f8e/engine/verification/match/engine.go#L504-L505

```golang
if isSystemChunk {
   // check that chunkDataPack.CollectionID is flow.ZeroID
   // VN knows how system chunk looks like: 
   //  * currently, system chunk is empty, i.e. we can check `collection.Len() == 0`
} else {
   // check that chunkDataPack.CollectionID is equal to the collection ID in the original block
}
```

I am worried about this check: `collection.Len() == 0`
To the best of my knowledge, the system chunk will contain transactions pretty soon (the transactions are protocol-determined instead; but they are still transactions I think). Therefore, I would suggest to create an issue: 
* include a factory for the system chunk which is shared between EN and VN
* VN uses the factory here to produce the collection that is supposed to be the system chunk (per protocol mandate). Subsequently, it checks that the `Checksum` of the expected system chunk collection matches the checksum of the collection provided by the EN.

@m4ksio do you have thoughts on this?